atd-vze/src/views/Users/User.js

```diff
@@ -1,40 +1,159 @@
-                      className="link"
-                    >
-                      <Button color="primary">
-                        <i className="fa fa-user-plus"></i> Edit User
-                      </Button>
-                    </Link>
-                  </Col>
-                </Row>
-              </CardBody>
-            </Card>
-          </Col>
-        </Row>
-      </div>
-    );
-  }
-}
+import React, { useState, useEffect } from "react";
+import { Link, Redirect, useParams } from "react-router-dom";
+import axios from "axios";
+import moment from "moment";
+import {
+  Button,
+  Card,
+  CardBody,
+  CardHeader,
+  Col,
+  Row,
+  Table,
+  Spinner,
+} from "reactstrap";
+
+const User = () => {
+  const token = window.localStorage.getItem("id_token");
+  const { id } = useParams();
+
+  // Define attributes to render in view and their labels and format for handling
+  const userAttributes = {
+    user_id: { label: "User ID", format: "string" },
+    name: { label: "Name", format: "string" },
+    email: { label: "Email", format: "string" },
+    app_metadata: { label: "Roles", format: "object", nestedKey: "roles" },
+    created_at: { label: "Created", format: "time" },
+    blocked: { label: "Blocked", format: "bool" },
+    last_login: { label: "Last login", format: "time" },
+    updated_at: { label: "Update at", format: "time" },
+    logins_count: { label: "Logins count", format: "string" },
+    last_ip: { label: "Last IP", format: "string" },
+  };
+
+  const [user, setUser] = useState(null);
+  const [isUserDeleted, setIsUserDeleted] = useState(false);
+  const [isUserBlocked, setIsUserBlocked] = useState(false);
+
+  useEffect(() => {
+    const endpoint = `${process.env.REACT_APP_CR3_API_DOMAIN}/user/get_user/${id}`;
+    axios
+      .get(endpoint, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      })
+      .then(res => {
+        // If user not found, redirect to users list
+        if (res.data.statusCode === 404) {
+          setIsUserDeleted(true);
+        } else {
+          setUser(res.data);
+          if (!!res.data.blocked) {
+            setIsUserBlocked(true);
+          }
+        }
+      });
+  }, [token, id]);
+
+  // Handle formats of data defined in userAttributes
+  const formatUserData = user =>
+    Object.entries(userAttributes).map(([key, value]) => {
+      const label = userAttributes[key].label;
+      let formattedValue = null;
+
+      const format = userAttributes[key].format;
+
+      if (format === "string") {
+        formattedValue = user[key];
+      } else if (format === "bool") {
+        formattedValue = !!user[key] ? "Yes" : "No";
+      } else if (format === "time") {
+        formattedValue = moment(user[key]).format("MM/DD/YYYY, h:mm:ss a");
+      } else if (format === "object") {
+        const nestedKey = value.nestedKey;
+        formattedValue = user[key][nestedKey].join(", ");
+      }
```

Comment:
consider using `switch` instead of `else if` for these types of blocks.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
👍I'll make this change before merging.